src/steps/check-pardot-tracking.ts

```diff
@@ -0,0 +1,80 @@
+import { isNullOrUndefined } from 'util';
+import { BaseStep, Field, StepInterface } from '../core/base-step';
+import { Step, RunStepResponse, FieldDefinition, StepDefinition } from '../proto/cog_pb';
+
+export class CheckPardotTrackingStep extends BaseStep implements StepInterface {
+
+  protected stepName: string = 'Check that Pardot tracking loads';
+  // tslint:disable-next-line:max-line-length
+  protected stepExpression: string = 'the tracking code for pardot account (?<aid>\\d+) and campaign (?<cid>\\d+) should have loaded';
+  protected stepType: StepDefinition.Type = StepDefinition.Type.VALIDATION;
+  protected expectedFields: Field[] = [{
+    field: 'aid',
+    type: FieldDefinition.Type.NUMERIC,
+    description: 'Pardot Account ID',
+  }, {
+    field: 'cid',
+    type: FieldDefinition.Type.NUMERIC,
+    description: 'Pardot Campaign ID',
+  }, {
+    field: 'customDomain',
+    type: FieldDefinition.Type.URL,
+    description: 'Custom Tracker Domain',
+    optionality: FieldDefinition.Optionality.OPTIONAL,
+  }, {
+    field: 'withParameters',
+    type: FieldDefinition.Type.MAP,
+    description: 'Parameter Checks, an optional map of query parameters and their expected values.',
+    optionality: FieldDefinition.Optionality.OPTIONAL,
+  }];
+
+  async executeStep(step: Step): Promise<RunStepResponse> {
+    const stepData: any = step.getData().toJavaScript();
+    const aid = stepData.aid;
+    const cid = stepData.cid;
+    const customDomain = stepData.customDomain || 'pi.pardot.com';
```

Comment:
- [ ] This fallback string should include the protocol (e.g. `https://pi.pardot.com`)

For context: the `FieldDefinition.Type.URL` type on the `customDomain` implies a fully qualified URL (including protocol).  If you run this step with `crank cog:steps` and enter a host without a protocol, I believe it won't let you continue because it is invalid.